compiler-rt/lib/plsan/plsan.cpp

```diff
@@ -30,46 +30,31 @@ extern "C" void __plsan_store(void **lhs, void *rhs) {
   __plsan::reference_count(lhs, rhs);
 }
 
-extern "C" LazyCheckInfo *__plsan_free_local_variable(void **arr_start_addr,
-                                                      uptr size, void *ret_addr,
-                                                      bool is_return,
-                                                      bool is_allocated) {
+extern "C" void __plsan_free_local_variable(void **arr_start_addr, uptr size,
+                                            void *ret_addr, bool is_return,
+                                            bool is_allocated) {
 
   if (!is_allocated)
-    return nullptr;
+    return;
 
-  __sanitizer::Vector<void *> *ref_count_zero_addrs =
-      __plsan::free_local_variable(arr_start_addr, size, ret_addr, is_return);
+  __plsan::free_local_variable(arr_start_addr, size, ret_addr, is_return);
 
   // This return will be changed. It have to contain stack trace data.
   // __builtin_return_address(0) will return program counter
-  LazyCheckInfo *lazy_check_info =
-      (LazyCheckInfo *)__sanitizer::InternalAlloc(sizeof(LazyCheckInfo));
-  CHECK(lazy_check_info);
-  lazy_check_info->RefCountZeroAddrs = ref_count_zero_addrs;
-  if (is_return) {
-    __sanitizer::InternalFree(lazy_check_info);
-    return nullptr;
-  } else {
-    return lazy_check_info;
-  }
 }
 
-extern "C" void __plsan_lazy_check(LazyCheckInfo *lazy_check_info,
-                                   void *ret_addr) {
+extern "C" void __plsan_lazy_check(void *ret_addr) {
   __sanitizer::Vector<void *> *lazy_check_addr_list =
-      lazy_check_info->RefCountZeroAddrs;
+      __plsan::local_var_ref_count_zero_list;
```

Comment:
I'm pretty sure that this commit will not work correctly.

How local_var_ref_count_zero_list can be used before it is defined?
the thread_local variable is introduced in the next commit, but this commit already uses it?

It doesn't make sense to me. Why not merge the last two commits into single one?

Reply:
I think you saw commit logs wrong order. local_var_ref_count_zero_list defined last commit.

Reply:
Oh, you are right!
My mistake.

Reply:
And do you think [68f1ff2](https://github.com/hygoni/precise-leak-sanitizer/pull/105/commits/68f1ff2558184936a8de680c8d4a64407283ab46), [a716e12](https://github.com/hygoni/precise-leak-sanitizer/pull/105/commits/a716e12a4ddebde875f8675482fa0d060b08da5c) these 2 commits should be merge?